keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1108 @@
+  * Complicated to implement as it requires additional logic in kubelet to monitor init containers completion after which it will set/change pod level cgroup settings.
+
+**Note: Option 1 seems easier to implement in the alpha phase to gather feedback from the users. If there are enough supporting use cases for Option 2, we can explore 
+the implementation in the beta phase of the KEP. Option 2 will also require discussions about dynamic QoS classes as per the execution phase of the pod 
+(init vs non-init phase).**
+
+
+#### Admission Controller
+
+* LimitRanger
+
+  This admission controller applies default resource requests to the Pods. In the existing implementation (without this KEP), it sums up container-level requests to calculate
+  effective pod requests. The logic in the LimitRanger controller needs to be changed to use pod-level requests for effective pod request calculation if pod-level resources 
+  specifications are set.
+
+* ResourceQuota
+
+  One of the use cases of the ResourceQuota admission controller is to limit the total sum of compute resources requests/limits for all pods in a given namespace. The ResourceQuota 
+  controller currently tracks resource requests/limits at pod level by aggregating container-level requests/limits. With pod-level resource specs, it should use pod-level requests/limits 
+  instead of aggregating container-level requests/limits to calculate effective pod resources requests/limits.
+
+
+#### Eviction Manager
+
+For memory.available signal, the eviction manager checks the pod's memory usage and compares it with memory requests at pod level to determine the pod eviction order. It aggregates the 
+container-level memory requests to calculate the effective pod-level memory request.When pod-level memory requests are specified, the Eviction Manager should use the pod-level request 
+directly instead of aggregating container-level requests. This logic should be modified to check if pod-level requests are set. If so, use the pod-level request; otherwise, fall back 
+to aggregating container requests.
+
+
+#### PodOverhead
+
+Since the PodOverhead feature works at pod-level cgroup, pod-level resource specification and PodOverhead will work together seamlessly. When the kube-scheduler is deciding 
+which node should run a new Pod, the scheduler considers that Pod's overhead as well as the sum of container requests for that Pod. Scheduler should add pod-level request,
+podoverhead to find a suitable node, when pod level request is specfied.
+
+Likewise the kubelet will need to add PodOverhead to the limits specified in the pod-level spec (instead of sum of container level requests) when setting the upper limit for
+the pod-level cgroup (i.e. cpu.max and memory.max).
+
+
+#### [Scoped for Beta] HugeTLB cgroup
+
+TBD. Do not review for the alpha stage.
+
+
+#### [Scoped for Beta] Topology Manager
+
+TBD. Do not review for the alpha stage.
```

Comment:
at least high level explanation on how this will be integrated is needed for alpha stage